docs/superpowers/plans/2026-05-30-elevation-mismatch-warning.md

```
@@ -0,0 +1,1240 @@
+
+```typescript
+import { setWarning, clearWarning } from "@iracedeck/deck-core";
+
+setWarning("elevation-mismatch", "warning", "…message…"); // upsert, keyed by id
+clearWarning("elevation-mismatch");                         // remove by id
+```
+
+Records are keyed by `id` so independent producers coexist. `setWarning` skips the write when an identical record already exists; `clearWarning` is a no-op when the id is absent. The `ird-warnings` PI web component (auto-injected by `head-common.ejs`) renders the array and prepends a per-level icon — so warning **messages must not start with their own emoji**. Banners are state-driven and not dismissible: a warning persists until its condition clears.
+
+Reference producer: the elevation-mismatch detector wired in both plugins' `plugin.ts` using `evaluateElevationWarning()` + `getElevationStatus()`.
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)